t/api/02-iso.t

```diff
@@ -966,4 +965,27 @@ subtest 'PUBLISH and STORE variables cannot include slashes' => sub {
     $schema->txn_rollback;
 };
 
+subtest 'Expand specified variables when scheduling iso' => sub {
+    $schema->txn_begin;
+    add_opensuse_test(
+        'test',
+        BUILD_HA            => '%BUILD%',
+        BUILD_SDK           => '%BUILD_HA%',
+        SHUTDOWN_NEEDS_AUTH => 1,
+        HDD_1   => '%DISTRI%-%VERSION%-%ARCH%-%BUILD_SDK%@%MACHINE%-minimal_with_sdk%BUILD_SDK%_installed.qcow2',
+        MACHINE => ['32bit', '64bit'],
+    );
+    my $res = schedule_iso({%iso, _GROUP_ID => '1002', TEST => 'test', BUILD => '176.6'}, 200);
+    is($res->json->{count}, 2, 'two job templates were scheduled');
+    $res = schedule_iso({%iso, _GROUP_ID => '1002', TEST => 'test', BUILD => '176.6', MACHINE => '64bit'}, 200);
+    is($res->json->{count}, 1, 'only the job template which machine is 64bit was scheduled');
+    my $result = $jobs->find($res->json->{ids}->[0])->settings_hash;
+    is(
+        $result->{HDD_1},
+        'opensuse-13.1-i586-176.6@64bit-minimal_with_sdk176.6_installed.qcow2',
+        'the specified variables were expanded correctly'
+    );
+    is($result->{BACKEND}, 'qemu', 'the BACKEND was added to settings correctly');
```

Comment:
Should we also check `MACHINE` explicitly here?

Reply:
Since `MACHINE` and `BACKEND` are the ones that could be overridden if the code didn't handle it correctly.

Reply:
`MACHINE` is checked in `HDD_1`.  `64bit` in `HDD_1`'s value is the `MACHINE`

Reply:
Ah, fair enough.